crates/orderbook/src/api/order_validation.rs

```diff
@@ -111,7 +111,11 @@ pub enum ValidationError {
     Partial(PartialValidationError),
     InsufficientFee,
     InsufficientFunds,
+    InsufficientAllowance,
     InvalidSignature,
+    // If fee and sell amount overflow u256
+    SellAmountOverflow,
```

Comment:
I would be OK with a generic 500-error in these cases (unless this actually is happening to our users).